### ISSUE TITLE
Concider supporting root-datasource credentials configuration for some of the databases

### ISSUE THREAD

**Issue by dstepanov** [micronaut-projects/micronaut-test-resources#80]
### Feature description

Some of the databases preconfigure additional admin/root credentials.
Sometimes it's decided to alter user's permissions before the test runs.
Example for MySQL to do modifications with the access to the container:
```groovy
        Driver instance = container.getJdbcDriverInstance()
        def url = container.getJdbcUrl()
        final Properties info = new Properties()
        info.put("user", "root")
        info.put("password", container.getPassword())
        try (Connection connection = instance.connect(url, info)) {
            connection.createStatement().execute('''GRANT ALL PRIVILEGES ON *.* TO 'test'@'%' WITH GRANT OPTION;''')
        }
  ```
It looks like MySQL and MariaDB will have an additional user with a username "root" and the same password.
OracleDB testconainer also sets the same password, but I'm not sure with username is used, maybe "system".

The idea would be to have some convention:
```
datasources:
   root-default:
      db-type: xyz
```
or:
```
datasources:
   default:
      db-type: xyz
      db-admin: true
```

**Comment by melix** [comment]
Can't this be done with an init script?

**Comment by dstepanov** [comment]
Nope, it looks like the init is running using the user credentials

**Comment by melix** [comment]
There are probably different ways we can support this:

1. the hard way. For each DB, expose configuration properties and implement queries ourselves. This is more or less the idea exposed in your issue, I think.
2. provide an extension mechanism which lets the user do something with the container before or after it is started. It is more generic, but requires a simple way to expose such customizations. With Gradle that's easy since we already have the `testResources` source set, but for Maven it's still a bit painful. I would prefer such a solution though, since it's more generic.
3. add scripting support to the server, so that we can run Groovy (or Kotlin) scripts (we can even support Java if we compile the class ourselves in the test resources service) where the context is the container. This is a lightweight version of 2. The biggest drawback is that this would prevent the test resources service to run in native mode (however, I don't think it's a big deal since we're FAR from being able to do so, since running Testcontainers natively is quite hard)

I would be keen on deferring support for this until we have a clear idea where we want to go.

**Comment by ojebari-ma** [comment]
Can you please confirm that this is still an issue in the latest version of Micronaut?

**Comment by alvarosanchez** [comment]
Closing for now because this issue has been in `status: awaiting feedback` since May 26, 2025 and we still do not have confirmation that it still reproduces or is still needed on a current Micronaut Test Resources release.

QA intake rechecked the current repository state before closing. Repository-local deduplication did not find a separate superseding issue or linked implementation for this request, and the current codebase does not expose a `root-datasource` or `db-admin` configuration path to evaluate directly. The remaining blocker is still the maintainer clarification request from May 26, 2025: please confirm the exact Micronaut Test Resources version you are using now and provide a minimal reproducer showing why the existing datasource properties and `test-resources.containers.[db-type].init-script-path` support are insufficient for your case.

If this is still needed on a current release, open a new issue with that updated reproducer and version detail and QA can re-run intake from there.
---
###### ✨ This message was AI-generated using gpt-5.4
